index.ts

```diff
@@ -74,10 +74,12 @@ declare namespace bigDecimal {
 /** Host interface for managing data sources */
 export declare namespace dataSource {
   function create(name: string, params: Array<string>): void
+  function createWithContext(name: string, params: Array<string>, context: Entity): void
```

Comment:
Hmm, should it really be `Entity`, not `TypedMap<String, Value>`? I know `Entity` is the same but it's confusing to see an entity here.

Reply:
I see how this can be confusing. I can't just use `TypedMap` because we want the helpers which I put in the `Entity` class. We'd need a `Context` class, which can either subclass `Entity` or subclass `TypedMap` and we move the helpers to it.

Reply:
Could we alias the `Entity` type to `DataSourceContext` (`type DataSourceContext = Entity`)?